### ISSUE TITLE
The vocab size definition for English and French Tokenizers is incorrect

### ISSUE THREAD

**Issue by sharan21** [thushv89/attention_keras#20]
The variables `en_vsize` and `fr_vsize` are dependant on the chosen size of the training set and therefore will vary for each test/train ratio. It is better to calculate these 2 variables by fitting the entire dataset into the tokenizer in order to compute vocab size for the entire dataset and then storing statically as a property of the given dataset

This avoids unseen word error while testing/validating the model and removes unnecessary vocab size dependancies.

**Comment by thushv89** [comment]
@sharan21 , 

What you're proposing is a very dangerous approach and results in data leakage. You are not supposed to know anything about the test data. Therefore, building vocabulary using training + test is wrong. It should only be the test set. To deal with unseen words, you can replace them with some common token or use [sub word approach](https://www.aclweb.org/anthology/P16-1162.pdf)
